pisound-btn/pisound-btn.c

```diff
@@ -330,42 +373,47 @@ static int get_action_script_path(enum action_e action, unsigned arg0, unsigned
 		strncpy(tmp, g_config_path, sizeof(tmp)-1);
 		dirname(tmp);
 		strncat(tmp, "/", sizeof(tmp)-1 - strlen(tmp));
-		strncat(tmp, script, sizeof(tmp)-1 - strlen(tmp));
+		strncat(tmp, action_script, sizeof(tmp)-1 - strlen(tmp));
 		tmp[sizeof(tmp)-1] = '\0';
 
-		strncpy(dst, tmp, n-1);
-		dst[n-1] = '\0';
+		strncpy(script, tmp, n-1);
+		script[n-1] = '\0';
 	}
 
 	strncpy(args, arguments, n-1);
 	args[n-1] = '\0';
 
-	return strlen(dst);
+	return strlen(script);
 }
 
-static void execute_action(enum action_e action, unsigned arg0, unsigned arg1)
+static void execute_action(enum action_e action, unsigned click_count, unsigned hold_time)
 {
 	char cmd[MAX_PATH_LENGTH + 64];
 	char arg[MAX_PATH_LENGTH + 64];
-	int n = get_action_script_path(action, arg0, arg1, cmd, arg, sizeof(cmd));
+	int n = get_action_script_path(action, click_count, hold_time, cmd, arg, sizeof(cmd));
 	if (n < 0)
 	{
 		fprintf(stderr, "execute_action: getting script path for action %u resulted in error %d!\n", action, n);
 		return;
 	}
 
+	if (n == 0)
+	{
+		fprintf(stderr, "execute_action: no command for action %s : click count %u hold time %u (%u seconds)\n", ACTION_NAME, click_count, hold_time, TICK_2_SECONDS(hold_time));
```

Comment:
This block won't be necessary once return value is fixed. The above line could be expanded to include more information like click & hold time.